src/probnum/diffeq/ode/ivp_examples.py

```diff
@@ -0,0 +1,464 @@
+        return fhn_rhs(t, y, params)
+
+    def jac(t, y):
+        return fhn_jac(t, y, params)
+
+    return IVP(timespan, initrv, rhs, jac)
+
+
+def fhn_rhs(t, y, params):
+    """RHS for FitzHugh-Nagumo model."""
+    y1, y2 = y
+    a, b, c, d = params
+    return np.array([y1 - y1 ** 3.0 / 3.0 - y2 + a, (y1 + b - c * y2) / d])
+
+
+def fhn_jac(t, y, params):
+    """Jacobian for FitzHugh-Nagumo model."""
+    y1, y2 = y
+    a, b, c, d = params
+    return np.array([[1.0 - y1 ** 2.0, -1.0], [1.0 / d, -c / d]])
+
+
+def lotkavolterra(timespan, initrv, params=(0.5, 0.05, 0.5, 0.05)):
+    r"""Initial value problem (IVP) based on the Lotka-Volterra model.
+
+    The Lotka-Volterra (LV) model is defined through
+
+    .. math::
+
+        f(t, y) =
+        \begin{pmatrix}
+            a y_1 - by_1y_2 \\
+            -c y_2 + d y_1 y_2
+        \end{pmatrix}
+
+    for some parameters :math:`(a, b, c, d)`.
+    Default is :math:`(a, b)=(0.5, 0.05, 0.5, 0.05)`.
+    This implementation includes the Jacobian :math:`J_f` of :math:`f`.
+
+    Parameters
+    ----------
+    timespan : (float, float)
+        Time span of IVP.
+    initrv : RandomVariable,
+        *(shape=(2, ))* -- Vector-valued RandomVariable that describes the belief
+        over the initial value. Usually its distribution is Constant (noise-free)
+        or Normal (noisy) with :math:`2`-dimensional mean vector and
+        :math:`2 \times 2`-dimensional covariance matrix.
+        To replicate "classical" initial values use the Constant distribution.
+    params : (float, float, float, float), optional
+        Parameters :math:`(a, b, c, d)` for the logistic IVP.
+        Default is :math:`(a, b, c, d)=(0.5, 0.05, 0.5, 0.05)`.
+
+    Returns
+    -------
+    IVP
+        IVP object describing the logistic IVP with the prescribed
+        configuration.
+    """
+
+    def rhs(t, y):
+        return lv_rhs(t, y, params)
+
+    def jac(t, y):
+        return lv_jac(t, y, params)
+
+    return IVP(timespan, initrv, rhs, jac)
+
+
+def lv_rhs(t, y, params):
+    """RHS for Lotka-Volterra."""
+    a, b, c, d = params
+    y1, y2 = y
+    return np.array([a * y1 - b * y1 * y2, -c * y2 + d * y1 * y2])
+
+
+def lv_jac(t, y, params):
+    """Jacobian for Lotka-Volterra."""
+    a, b, c, d = params
+    y1, y2 = y
+    return np.array([[a - b * y2, -b * y1], [d * y2, -c + d * y1]])
+
+
+def seir(timespan, initrv, params=(0.3, 0.3, 0.1, 1e7)):
+    r"""Initial value problem (IVP) based on the SEIR model.
+
+    The SEIR model with no vital dynamics is defined through
+
+    .. math::
+
+        f(t, y) =
+        \begin{pmatrix}
+            \frac{-\beta y_1 y_3}{N} \\
+            \frac{\beta y_1 y_3}{N} - \alpha y_2 \\
+            \alpha y_2 - \gamma y_3 \\
+            \gamma y_3
+        \end{pmatrix}
+
+    for some parameters :math:`(\alpha, \beta, \gamma, N)`.
+    N is a constant population count.
+    Default is :math:`(\alpha, \beta, \gamma, N)=(0.3, 0.3, 0.1, 10^7)`.
```

Comment:
isn't it that N is defined by the sum of values in initrv? Why is it defined as a parameter?

Reply:
and isnt there a better name than N? E.g. `population_count` or something?

Reply:
True. It would be sensible to save this value somewhere, though. Otherwise, the same value `population_count` would have to be computed in each evaluation of the `rhs`. So either we compute it in each evaluation, we provide it as a parameter (and give it a useful name), or we add it as an attribute to the `IVP` object, which would be maybe the fastest, yet somehow "hackiest" solution. What do you prefer?

Reply:
how about reading N off the initial random variable into a parameter, which is then fixed by
```python
    def rhs(t, y):
        return seir_rhs(t, y, params) 
```
anyway? I am not sure what the best solution here is, though passing an explicit parameter N that might contradict the initial state (and subsequently, all other states) seems error prone

Reply:
Done ✅